### PR TITLE
[fix][doc] Sync recent changes on versioned docs

### PR DESCRIPTION
**Pull request by momo-jun**
### Modifications

* PIP-190: Sync recent doc changes to 2.8.x/2.9.x/2.10.x docs during the switchover time window of https://github.com/apache/pulsar/pull/17074.
   * #17030
   * #17039
   * #16315
   * #17057
* PIP-190: Update wiki/release-process to remove the doc set generation for bug-fix releases.
* Fix https://github.com/apache/pulsar/issues/17119.
* Add a link to release notes to sidebar navigation.
image

### Documentation

- [x] `doc`

### CODE REVIEW

File: wiki/release/release-process.md [hunk 1]
```diff
@@ -521,21 +523,11 @@ After you run this command, a new folder `version-<release-version>` is added in
...
-### Update the site for minor releases
-
-The new updates for the minor release docs are processed in its doc set and sidebar file directly before release. You can follow step 4~8 (in major release) to update the site. You'll also need to add this new version to `versions.json`.
-
-To make preparation for the next minor release, you need to generate the doc set and sidebar file based on the previous release. Take `2.X.2` as example, `2.X.2` doc set and sidebar.json file are generated based on `2.X.1`. You can make it with the following steps:
-
-1. Copy the `version-2.X.1` doc set and `version-2.X.1-sidebars.json` file and rename them as `version-2.X.2` doc set and `version-2.X.2-sidebars.json` file.
-
-2. Update the "id" from `version-2.X.1` to `version-2.X.2` for the md files in the `version-2.X.2` doc set and `version-2.X.2-sidebars.json` file.
+:::note
```
Comment (Anonymitaet) [inline wiki/release/release-process.md:526]:
```suggestion
> **Note**
```
This is a .md file

File: wiki/release/release-process.md [hunk 2]
```diff
@@ -521,21 +523,11 @@ After you run this command, a new folder `version-<release-version>` is added in
...
 
-3. Submit the new doc set and sidebar.json file to the apache/pulsar repository.
+Starting from 2.8, you don't need to generate the doc set or update the Pulsar site for bug-fix releases, such as 2.8.1, 2.8.2 and so on. The generic doc set 2.8.x is used.
```
Comment (Anonymitaet) [inline wiki/release/release-process.md:528]:
```suggestion
> Starting from 2.8, you don't need to generate an independent doc set for each release or update the Pulsar site for bug-fix releases, such as 2.8.1, 2.8.2, and so on. The generic doc set 2.8.x is used.
```

File: wiki/release/release-process.md [hunk 3]
```diff
@@ -521,21 +523,11 @@ After you run this command, a new folder `version-<release-version>` is added in
...
 
-> **Note**
-> - The `yarn run version <release-version>` command generates the new doc set and sidebar.json file based on the `site2/docs` folder.
-> - The minor release doc is generated based on the previous minor release (e.g.: `2.X.2` doc is generated based on `2.X.1`, and `2.X.3`doc is generated based on `2.X.2`), so you cannot use the `yarn run version <release-version>` command directly.
+:::
```
Comment (Anonymitaet) [inline wiki/release/release-process.md:530]:
```suggestion

```

**Review by Anonymitaet — APPROVED** [review]

**Review by nodece — APPROVED** [review]

File: wiki/release/release-process.md [hunk 4]
```diff
@@ -504,38 +502,28 @@ After you run this command, a new folder `version-<release-version>` is added in
...
-
-3. Submit the new doc set and sidebar.json file to the apache/pulsar repository.
+9. Generate the doc set and sidebar file for the next minor release `2.X.x` based on the `site2/docs` folder. You can follow steps 1, 2, and 3, and submit those files to the `apache/pulsar` repository. This step is a preparation for the `2.X.x` release.
 
 > **Note**
-> - The `yarn run version <release-version>` command generates the new doc set and sidebar.json file based on the `site2/docs` folder.
-> - The minor release doc is generated based on the previous minor release (e.g.: `2.X.2` doc is generated based on `2.X.1`, and `2.X.3`doc is generated based on `2.X.2`), so you cannot use the `yarn run version <release-version>` command directly.
+> Starting from 2.8, you don't need to generate an independent doc set or update the Pulsar site for bug-fix releases, such as 2.8.1, 2.8.2, and so on. Instead, the generic doc set 2.8.x is used.
+
+:::
```
Comment (Anonymitaet) [inline wiki/release/release-process.md:526]: image

@momo-jun can you fix this layout issue？

Comment (momo-jun) [reply wiki/release/release-process.md:526]: Thanks for the finding. I will fix it in a follow-up PR.

File: wiki/release/release-process.md [hunk 5]
```diff
@@ -473,18 +473,16 @@ Steps and examples see [Pulsar Release Notes Guide](https://docs.google.com/docu
 
 ## 17. Update the site
 
-The workflow for updating the site is slightly different for major and minor releases.
-
-### Update the site for major releases
-For major release, the website is updated based on the `master` branch.
+### Update the site for minor releases
+For minor releases, such as 2.10, the website is updated based on the `master` branch.
```
Comment (BewareMyPower) [inline wiki/release/release-process.md:477]: @momo-jun Why did you make this change? I think only major releases (like 2.10.0) are updated based on the `master` branch.

ping @mattisonchao @Technoboy- @codelipenghui @gaoran10

Comment (BewareMyPower) [reply wiki/release/release-process.md:477]: I need to make sure I didn't understand wrong.
- major release: 2.X.0 (checkout from `master`)
- minor release: 2.X.Y (checkout from `branch-2.X`)

Comment (momo-jun) [reply wiki/release/release-process.md:477]: @BewareMyPower This change is part of [PIP-190](https://github.com/apache/pulsar/issues/16637). 

You are right that 2.10 is updated based on the master branch. Quite often we regard it as a major version in daily talk. But it should be a minor version, while 3.0 is the next major version. 
`Pulsar version number is comprised of 3 components: major.minor.bug-fix (patch)` according to the [definition in PIP-47](https://github.com/apache/pulsar/wiki/PIP-47%3A-Time-Based-Release-Plan#what-about-version-numbers).

Comment (momo-jun) [reply wiki/release/release-process.md:477]: As clarified by @codelipenghui and @BewareMyPower, the current consensus in the Pulsar community is:
* feature release (major release): 2.10.0, 2.11.0...
* bug-fix release (minor release): 2.10.1, 2.10.2...
I will update the wording `minor` accordingly.

**Pull request merged**